src/main/java/net/neoforged/neoforge/fluids/capability/IFluidHandler.java

```diff
@@ -6,15 +6,30 @@
 package net.neoforged.neoforge.fluids.capability;
 
 import net.neoforged.neoforge.fluids.FluidStack;
-import net.neoforged.neoforge.fluids.IFluidTank;
+import net.neoforged.neoforge.transfer.ResourceHandlerDeprecationHandling;
+import net.neoforged.neoforge.transfer.ResourceHandlerUtil;
+import net.neoforged.neoforge.transfer.TransferAction;
+import net.neoforged.neoforge.transfer.handlers.resources.IResourceHandler;
+import net.neoforged.neoforge.transfer.resources.FluidResource;
+import net.neoforged.neoforge.transfer.resources.IResource;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
 
 /**
- * Implement this interface as a capability which should handle fluids, generally storing them in
- * one or more internal {@link IFluidTank} objects.
- * <p>
- * A reference implementation is provided {@link TileFluidHandler}.
+ * Implement this interface as a capability which should handle fluids
+ *
+ * @deprecated Use {@link IResourceHandler} bound by type {@link FluidResource} instead.
+ * 
+ *             <pre>{@code
+ *  IResourceHandler<FluidResource>
+ * }</pre>
  */
+
+@Deprecated(since = ResourceHandlerDeprecationHandling.MC_1_21_6, forRemoval = true)
 public interface IFluidHandler {
```

Comment:
There should be an implementation of `IFluidHandler` that wraps an `IResourceHandler<FluidResource>` to help with migration at the use sites of the capability.